### PR TITLE
[Jetpack Focus] Add Jetpack banner to Jetpack Scan view

### PR DESCRIPTION
**Pull request by twstokes**
Part of https://github.com/wordpress-mobile/WordPress-iOS/issues/19448

## Description
- Adds the Jetpack banner to the Jetpack Scan view in the WordPress app

| Jetpack Scan | Banner should not obscure toast |
| - | - |
| image | image |

**Note: There are currently no banner or badges for subviews under the History section.**

| History button | History view |
| - | - |
| image | Simulator Screen Shot - iPhone 14 - 2023-01-05 at 22 19 41 |

## Testing

**Prerequisite:** Access to a site that has Jetpack Scan enabled.

### Banner appears in WordPress
1. Load the WordPress app.
2. From My Site, tap Scan in the Jetpack section.
3. **Expect:** The Jetpack banner to appear at the bottom of the screen.
4. **Expect:** Tapping on the Jetpack banner shows the Jetpack overlay.
5. **Expect:** The event `🔵 Tracked: jetpack_powered_banner_tapped <screen: scan>` to be fired.

### Banner does not appear in Jetpack
1. Load the Jetpack app.
2. From My Site, tap Scan in the Jetpack section.
3. **Expect:** No Jetpack banner to be shown.

### Banner visibility behavior
- Initially the banner should be shown when the screen appears.
- Scrolling down hides the banner and the banner reappears only when the view is scrolled to the top.
- If there isn't enough content to scroll, the banner doesn't show / hide on scroll.

### Items to test:
- Light / Dark mode
- Dynamic Type / a11y
- Different screen orientations
- iPhone / iPad

## Regression Notes
1. Potential unintended areas of impact
    - Top navigation bar buttons / functionality.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
    - Manual testing above.

3. What automated tests I added (or what prevented me from doing so)
    - None.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by twstokes** [comment]
@hassaanelgarem I've realized that this view will also have similar issues with the navigation item not being updated by the child. "History" won't appear until I fix that.

**Review by hassaanelgarem — COMMENTED** [review]
> I've realized that this view will also have similar issues with the navigation item not being updated by the child. "History" won't appear until I fix that.

Makes sense 👍 

I've tested the rest of the functionality, and it works as expected.

File: WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m [hunk 1]
```diff
@@ -1593,7 +1593,7 @@ - (void)showActivity
 
 - (void)showScan
 {
-    JetpackScanViewController *controller = [[JetpackScanViewController alloc] initWithBlog:self.blog];
+    UIViewController *controller = [JetpackScanViewController withJPBannerForBlog:self.blog];
     controller.navigationItem.largeTitleDisplayMode = UINavigationItemLargeTitleDisplayModeNever;
```
Comment (twstokes) [inline WordPress/Classes/ViewRelated/Blog/Blog Details/BlogDetailsViewController.m]: TODO: Move this.

**Review by hassaanelgarem — APPROVED** [review]
History button is showing and working properly now. And I also retested the main scenarios and all's working as expected 🚀

**Pull request merged**